### ISSUE TITLE
extend MA0110 

### ISSUE THREAD

**Issue by EngRajabi** [meziantou/Meziantou.Analyzer#334]
Currently, the MA0110 feature only works if we use it this way.

```csharp
new Regex("constant pattern"); // report diagnostic
```

It does not work if we use regex this way
```csharp
private static readonly Regex NumberRegex = new Regex(@"[\d-]", regexOptions, TimeSpan.FromMilliseconds(50));
```

**Comment by meziantou** [comment]
The second sample cannot be replaced by a source generated regex as the `regexOptions` is not a constant (it seems to be a reference to a field?). Could you provide the full code, so I can understand the context?

**Comment by EngRajabi** [comment]
```csharp
    private static readonly RegexOptions regexOptions =
        RegexOptions.Multiline | RegexOptions.Compiled | RegexOptions.IgnoreCase;
```

**Comment by meziantou** [comment]
As I said, only constant values can be used for the attribute. `regexOptions` is not a constant (`static readonly`), so MA0110 doesn't report it.
